client/van_connector_create_test.go

```diff
@@ -105,11 +102,14 @@ func TestConnectorCreateInterior(t *testing.T) {
 		})
 		assert.Check(t, err, "Unable to create connector")
 
-		// TODO: make more deterministic
-		time.Sleep(time.Second * 1)
-		assert.Assert(t, cmp.Equal(c.secretsExpected, secretsFound, trans), c.doc)
-	}
+		select {
```

Comment:
Thanks you for the proposal. While I am not a fan of the time.sleep, the reason I did not use a channel was to have a general pattern for detecting negative results (e.g. to detect unintential creation of resources). While that may not apply so much to the connector tests, it gets uses in other tests where multiple resources are expected. It would be good to discuss whether this is a reasonable pattern for the unit tests or if there is a better way to detect negative results.

Reply:
Aha!, thanks for the feedback, will take a look to those other tests, and think about this, basically I think we should try to reduce testing time, somehow, ideally taking 1s to run all unit-tests, not ideally ... no more than 10 ... ?